app/models/ext_management_system.rb

```diff
@@ -426,6 +426,24 @@ def enable!
     update!(:enabled => true)
   end
 
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    workers = MiqWorker.find_current_or_starting.where(:queue_name => "ems_#{id}") #better way to get 'queue_name'?
+    if workers.count.zero?
+      destroy_queue # from AsyncDeleteMixin
+    else
+      # Something like a 15 seconds before trying again?
+      MiqQueue.put(
+          :deliver_on  => 15.seconds.from_now,
+          :class_name  => self.class.name,
+          :instance_id => id,
+          :method_name => "orchestrate_destroy"
```

Comment:
> How often do this happen in practice? Can we just wait 5 minutes between checks?

@jrafanie I think users expect for the deletion process to be fairly quick. Can we instead cancel working jobs? can we do that safely?